### ISSUE TITLE
Failing to negotiate TLS on reinvite

### ISSUE THREAD

**Issue by davehorton** [sipwise/rtpengine#1245]
I apologize in advance, as I am sure I am doing something wrong here.  Perhaps you could spot it.

The scenario is this: I have a B2BUA receiving RTP INVITEs from a freeswitch server in my network, and sending SRTP to a webrtc (jssip) client on the internet.

I have configured my rtpengine to have a 'public' and a 'private' interface.  The webrtc client is reached on the public, the freeswitch on the private.

I set up an initial call successfully from freeswitch to the webrtc client, media passing through rtpengine nicely.

The webrtc client (the called party) then goes on hold, sending me a re-invite with a:sendonly.

Here is how I am handling that reinvite:
1. I am calling 'offer' on rtpengine with the SDP in the reinvite.
2. I am taking the sdp returned from rtpengine and sending it in a reinvite to freeswitch
3. I am taking the sdp in the 200 ok from freeswitch and calling 'answer' on rtpengine
4. I am taking the sdp returned from rtpengine and sending it in a 200 OK to the webrtc client.

Now, I am wondering first of all if that is the correct process.  Because I notice that rtpengine changes UDP ports for both media sessions (that pointing towards the freeswitch as well as that pointing towards the webrtc client), and I was not expecting that.

The further problem is that after this re-invite transaction is concluded, jssip immediately sends a BYE and logs this message:

"JsSIP:ERROR:RTCSession emit "peerconnection:setremotedescriptionfailed" [error:DOMException: Failed to execute 'setRemoteDescription' on 'RTCPeerConnection': Failed to set remote answer sdp: Failed to apply the description for m= section with mid='1': Failed to set SSL role for the transport."

[rtpengine log](https://gist.github.com/davehorton/d38e99827b159186c6f243498eb80e5c)

[annotated sip trace](https://gist.github.com/davehorton/7d5553e786c4b07107ffcc34f5ca25ee)

I'd like to understand if I can have rtpengine not change ports during this transaction (though I suspect this is probably related to whatever I am doing wrong).  I am also wondering if I need to modify the tls negotiation -- currently on the 200 OK to the re-invite the SDP from rtpengine indicates that it will take the active role - whereas on the initial INVITE to the webrtc client, the client indicated it was taking the active role.  Not sure if I need to change this..

Any help would be appreciated.

I am using rtpengine Version 8.5.4.0+0~mr8.5.4.0

**Comment by rfuchs** [comment]
What's happening is that the direction of the re-invite is reversed. The original offer from FS lists `"from-tag": "UUKF63KS2e06H"` so this tag identifies the FS endpoint. The answer then lists `"to-tag": "49kraktqtv"` so this tag identifies the WebRTC endpoint.

The re-invite is coming from the WebRTC endpoint and going to FS, so it should use `49kraktqtv` as from-tag and `UUKF63KS2e06H` as to-tag. However, the log shows the opposite: `"from-tag": "UUKF63KS2e06H", ... "to-tag": "49kraktqtv"` and so from rtpengine's POV this re-invite seems to be going from FS to the WebRTC endpoint, which then includes changed SDP attributes, changed IP/port, etc. The tags in the answer are also reversed, and this is why the WebRTC endpoint doesn't like the SDP (it's made for the FS side of the call).

TLDR: swap from/to tags in the re-invite offer/answer and you should be good.

**Comment by davehorton** [comment]
This worked, thanks again!
